crates/api_common/src/request.rs

```diff
@@ -105,7 +105,11 @@ pub fn generate_post_link_metadata(
     }
     // Generate local thumbnail if allowed
     else if allow_generate_thumbnail {
-      match post.url.or(metadata.opengraph_data.image) {
+      match post
+        .url
+        .filter(|_| is_image_post)
```

Comment:
This is fine for now, but I argued earlier that we should've moved the entire `is_image_post` check up above, since its checked in almost every branch now anyway.